File: c/tree.c
#include "tree.h"

#include <stdlib.h>

TreeNode *newTreeNode(int val) {
    TreeNode *node = malloc(sizeof(TreeNode));
    if (node) {
        node->val = val;
        node->left = node->right = NULL;
    }
    return node;
}

File: Python/tree.py
from typing import Optional


class TreeNode:
    def __init__(self, val: int):
        self.val: int = val
        self.left: Optional[TreeNode] = None
        self.right: Optional[TreeNode] = None

    def __eq__(self, other: object) -> bool:
        if self is other:
            return True
        return (isinstance(other, TreeNode)
                and self.val == other.val
                and self.left == other.left
                and self.right == other.right)
